lib/item_stream_handler.rb

```diff
@@ -13,81 +14,85 @@ def avro_decoder(name)
     @avro_decoders[name]
   end
 
-  def add_checkout(checkout)
-    @checkouts = [] if @checkouts.nil?
-
-    # Application.logger.debug "Adding checkout #{checkout}"
-
-    # Add checkout to end:
-    @checkouts << checkout
-
-    Application.logger.debug "Collected checkouts size is now #{@checkouts.size}"
-
-    # Make sure @checkouts doesn't grow behond max:
-    @checkouts = constrain_size @checkouts, MAX_CHECKOUTS_IN_MEMORY
-  end
-
   # Reduce array to the given size
   # e.g. constrain_size([1, 2, 3], 2) => [2, 3]
   def constrain_size(arr, size)
-    excess_records = arr.size - size
-    arr.shift excess_records if excess_records > 0
-    arr
+    arr[(arr.size - size) ... size ]
   end
 
-  def update_tally_if_necessary
+  def clear_tally_if_necessary
     if Time.now.day != ItemTypeTally[:time].day
       ItemTypeTally[:time] = Time.now
       ItemTypeTally[:tallies] = Hash.new {|h,k| h[k] = 0}
     end
   end
 
+
+  def remove_old_ids(id_hash)
+    id_hash.each do |(id, time)|
+      id_hash.delete(id) if Time.now - time > ENV["CHECKOUT_ID_EXPIRE_TIME"].to_i
+    end
+  end
+
+  def cleanup
```

Comment:
I feel this name is a bit too ambiguous

Reply:
hmmm... what about `clear_out_of_date_data`? I'm happy to go with whatever you suggest

Reply:
`clear_out_of_date_records` or `clear_out_of_date_items`?